src/forms/SpatialBoundsField.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from "react";
+import { getBoundsFromWmsLayer } from "../utils/getBoundsFromGeoServer";
+import ClearInputButton from "./ClearInputButton";
+import styles from "./DurationField.css";
+import formStyles from "../styles/Forms.css";
+import inputStyles from "../styles/Input.css";
+import thresholdsStyles from './ThresholdsSelection.css';
+
+interface SpatialBoundsProps {
+    value: {
+        north: number,
+        east: number,
+        south: number,
+        west: number
+    } | null,
+    otherValues: {
+        wmsLayerName: string,
+        wmsLayerSlug: string,
+        wmsLayerUrl: string,
+    },
+    valueChanged: Function,
+};
+
+// Validator
+export const spatialBoundsValidator = (fieldValue: SpatialBoundsProps['value']) => {
+    if (fieldValue) {
+        const {
+            north,
+            east,
+            south,
+            west
+        } = fieldValue;
+
+        if (
+            !north ||
+            !east ||
+            !south ||
+            !west
+        ) {
+            return "Please enter a number in all fields or clear all inputs"
+        } else if (north < south) {
+            return "North coordinate cannot be smaller than South coordinate"
+        } else if (east < west) {
+            return "East coordinate cannot be smaller than West coordinate"
+        } else {
+            return false;
+        };
+    };
+};
+
+export default class SpatialBoundsField extends Component<SpatialBoundsProps, {}> {
+    updateSpatialBounds(key: string, value: string) {
+        this.props.valueChanged({
+            ...this.props.value,
+            [key]: parseFloat(value)
+        });
+    }
+    removeSpatialBounds() {
+        this.props.valueChanged(null);
+    }
+    render() {
+        const {
+            value,
+            otherValues,
+            valueChanged,
+        } = this.props;
+
+        const {
+            wmsLayerSlug,
+            wmsLayerUrl,
+        } = otherValues;
+
+        let north, east, south, west;
+
+        if (value) {
+            north = value.north ? value.north : "";
+            east = value.east ? value.east : "";
+            south = value.south ? value.south : "";
+            west = value.west ? value.west : "";
```

Comment:
Same here -- 0 is valid.

Reply:
same here, I changed the check to `Number.isFinite(value.north) ? value.north : "";`